applicationinsights-react-js/src/ReactPlugin.ts

```diff
@@ -25,30 +33,51 @@ export default class ReactPlugin extends BaseTelemetryPlugin {
                 super.initialize(config, core, extensions, pluginChain);
-                _extensionConfig =
-                    config.extensionConfig && config.extensionConfig[_self.identifier]
-                        ? (config.extensionConfig[_self.identifier] as IReactExtensionConfig)
-                        : { history: null };
-        
-                arrForEach(extensions, ext => {
-                    const identifier = (ext as ITelemetryPlugin).identifier;
-                    if (identifier === 'ApplicationInsightsAnalytics') {
-                        _analyticsPlugin = (ext as any) as IAppInsights;
+
+                _self._addHook(onConfigChange(config, (details) => {
+                    let ctx = _self._getTelCtx();
+                    _extensionConfig = ctx.getExtCfg<IReactExtensionConfig>(this.identifier, defaultReactExtensionConfig);
+                    _analyticsPlugin = core.getPlugin<any>(AnalyticsPluginIdentifier)?.plugin as IAppInsights;
+                    
+                    if (isFunction(_unlisten)) {
+                        _unlisten();
                     }
-                });
-                if (_extensionConfig.history) {
-                    _addHistoryListener(_extensionConfig.history);
-                    const pageViewTelemetry: IPageViewTelemetry = {
-                        uri: _extensionConfig.history.location.pathname
-                    };
-                    _self.trackPageView(pageViewTelemetry);
-                }
+                    if (_extensionConfig.history) {
+                        _addHistoryListener(_extensionConfig.history);
+                        if (!_pageViewTracked){
+                            const pageViewTelemetry: IPageViewTelemetry = {
+                                uri: _extensionConfig.history.location.pathname
+                            };
+                            _self.trackPageView(pageViewTelemetry);
+                            _pageViewTracked = true;
+                        }
+                    }
+
+                    // if (_prevHistory != _extensionConfig.history){
+                    //     if (isFunction(_unlisten)) {
+                    //         console.log("unlisten");
+                    //         _unlisten();
+                    //         _unlisten = null;
+                    //     }
+                    //     if (_extensionConfig.history){
+                    //         _addHistoryListener(_extensionConfig.history);
+                    //         const pageViewTelemetry: IPageViewTelemetry = {
+                    //             uri: _extensionConfig.history.location.pathname
+                    //         };
+                    //         console.log("me track", _extensionConfig.history.location.pathname)
+                    //         _self.trackPageView(pageViewTelemetry);
```

Comment:
if the prev history added again, do we want to trackPageView again?

Reply:
No, I don't think so. Lets only do this on first initialization